src/lib.rs

```diff
@@ -1573,8 +1573,10 @@ impl Build {
                         if let Some(arch) =
                             map_darwin_target_from_rust_to_compiler_architecture(target)
                         {
+                            let deployment_target = env::var("IPHONEOS_DEPLOYMENT_TARGET")
```

Comment:
I'm not sure whether it is a good way. But as an iOS developer, I believe it looks good. Because in Xcode, Mac Catalyst minimal version is highly related to minimal iOS version. For example, if I choose iOS 14.0, I can only pick macOS 11+ for this situation:

<img width="640" alt="image" src="https://user-images.githubusercontent.com/2396817/166129562-9683a885-11ca-43b2-aa20-e9054e2a2c88.png">

<img width="636" alt="image" src="https://user-images.githubusercontent.com/2396817/166129584-75cced66-a890-460a-a84f-a4a26d2ed47e.png">